optaplanner-core/src/main/java/org/optaplanner/core/impl/localsearch/decider/acceptor/greatdeluge/GreatDelugeAcceptor.java

```diff
@@ -0,0 +1,93 @@
+
+public class GreatDelugeAcceptor extends AbstractAcceptor {
+
+    private Score initialWaterLevel = null;
+    private Score currentWaterLevel = null;
+
+    private Score waterLevelIncrementScore = null;
+    private Double waterLevelIncrementRatio = DEFAULT_WATER_LEVEL_INCREMENT_RATIO;
+
+    // Good value to come out from. Source: https://github.com/UniTime/cpsolver from Tomas Muller
+    private static final double DEFAULT_WATER_LEVEL_INCREMENT_RATIO = 0.00_000_005;
+
+
+    public Score getWaterLevelIncrementScore() {
+        return this.waterLevelIncrementScore;
+    }
+
+    public void setWaterLevelIncrementScore(Score waterLevelIncrementScore) {
+        this.waterLevelIncrementScore = waterLevelIncrementScore;
+        this.waterLevelIncrementRatio = null;
+    }
+
+    public Score getInitialWaterLevel() {
+        return this.initialWaterLevel;
+    }
+
+    public void setInitialWaterLevel(Score initialLevel) {
+        this.initialWaterLevel = initialLevel;
+    }
+
+    public Double getWaterLevelIncrementRatio() {
+        return this.waterLevelIncrementRatio;
+    }
+
+    public void setWaterLevelIncrementRatio(Double waterLevelIncrementRatio) {
+        this.waterLevelIncrementRatio = waterLevelIncrementRatio;
+    }
+
+    @Override
+    public void phaseStarted(LocalSearchPhaseScope phaseScope) {
+        super.phaseStarted(phaseScope);
+
+        if (initialWaterLevel != null) {
+            for (double initialLevelLevel : ScoreUtils.extractLevelDoubles(initialWaterLevel)) {
+                if (initialLevelLevel < 0.0) {
+                    throw new IllegalArgumentException("The initial level (" + initialWaterLevel
+                                                               + ") cannot have negative level (" + initialLevelLevel + ").");
+                }
+            }
+            currentWaterLevel = initialWaterLevel;
+
+        } else {
+            currentWaterLevel = phaseScope.getBestScore().negate();
+        }
+    }
+
+    @Override
+    public void phaseEnded(LocalSearchPhaseScope phaseScope) {
+        super.phaseEnded(phaseScope);
+        initialWaterLevel = null;
+        waterLevelIncrementRatio = DEFAULT_WATER_LEVEL_INCREMENT_RATIO;
+        waterLevelIncrementScore = null;
+    }
+
+    @Override
+    public boolean isAccepted(LocalSearchMoveScope moveScope) {
+
+        Score moveScore = moveScope.getScore();
+
+        return moveScore.compareTo(currentWaterLevel.negate()) >= 0;
+    }
+
+    @Override
+    public void stepEnded(LocalSearchStepScope stepScope) {
+        super.stepEnded(stepScope);
+
+        if (waterLevelIncrementScore != null) {
+            currentWaterLevel = currentWaterLevel.subtract(waterLevelIncrementScore);
+        } else {
+            Score increment = currentWaterLevel.multiply(waterLevelIncrementRatio);
+            currentWaterLevel = currentWaterLevel.subtract(increment);
+        }
+        logger.debug(currentWaterLevel.toShortString());
```

Comment:
@BorisKocian I think that this debug line needs to go away.

Reply:
I'm sorry, my bad.